python/dolfinx/nls/petsc.py

```diff
@@ -72,3 +98,432 @@ def setP(self, P: types.FunctionType, Pmat: PETSc.Mat):  # type: ignore
 
         """
         super().setP(P, Pmat)
+
+
+def create_snes_matrices_and_vectors(
```

Comment:
@michalhabera Questions the need for this in the public API - should users just use `create_*` directly?